### DISCUSSION TITLE
Move abort in 0x1::transaction_validation: 0x203ed

### DISCUSSION THREAD

**Discussion by 0xmaayan** [aptos-labs/aptos-developer-discussions#136] [Questions] [answered]
From TG

What could be the reason for `Move abort in 0x1::transaction_validation: 0x203ed` failure?

full code:
```
const transaction = await aptos.transaction.build.simple({
    sender: accountPrevious.accountAddress,
    data: {
        function: "0x1::aptos_account::transfer",
        typeArguments: [],
        functionArguments: [account.accountAddress, value],
    },
    options: {
        maxGasAmount: 1100
    }
});
const committedTxn = await aptos.signAndSubmitTransaction({ signer: accountPrevious, transaction: transaction });
```

https://explorer.aptoslabs.com/txn/0xa5d309f77d9d6b3d8faa7c521795fedea1d7b624e325c86239f15e5973bda641/userTxnOverview?network=mainnet

**Comment by BriungRi** [comment, marked as answer]
0x203ed translates the error into

0x2: [`OUT_OF_RANGE`](https://github.com/aptos-labs/aptos-core/blob/d763a7ad3405c37eba4c73c33e82b108f294635c/third_party/move/move-stdlib/sources/error.move#L28C9-L28C21)

3ed: [`PROLOGUE_ECANT_PAY_GAS_DEPOSIT`](https://github.com/aptos-labs/aptos-core/blob/d763a7ad3405c37eba4c73c33e82b108f294635c/aptos-move/framework/aptos-framework/sources/transaction_validation.move#L43)

Error is thrown [here](https://github.com/aptos-labs/aptos-core/blob/d763a7ad3405c37eba4c73c33e82b108f294635c/aptos-move/framework/aptos-framework/sources/transaction_validation.move#L279)

which is checking that the user's APT balance can pay the transaction fee (which it looks like it cannot)

**Reply by oleksiivinogradov** [reply]
And how to rewrite my code to pass?
I do: 
let value = parseInt(await balanceForAptosWalletWEI(prevousWallet)) - 1100;

and then 
const transaction = await aptos.transaction.build.simple({
    sender: accountPrevious.accountAddress,
    data: {
        function: "0x1::aptos_account::transfer",
        typeArguments: [],
        functionArguments: [account.accountAddress, value],
    },
    options: {
        maxGasAmount: 1100
    }
});

when I try to send tnx from regular wallet 1005 is enough to cover all and tnx passed. I try to change 1100 to more like 1600 but still no success

**Reply by gregnazario** [reply]
This can be variable based on what's going on in the transaction.  Sending to a non-existent account will cost more gas than an account that already exists (because it creates the account)

**Comment by 0xmaayan** [comment]
looks like that is error 1005, canot pay gas deposit, you are lack sufficient apt to cover the proposed max transaction fee
https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/transaction_validation.move#L43

Can follow this guide on how to debug an error code
https://github.com/aptos-labs/aptos-developer-discussions/discussions/103#discussioncomment-8910272
